utilities/balancing-torque-control/script/blf-balancing-torque-control.py

```diff
@@ -0,0 +1,509 @@
+            wrench=left_wrench, contact_force_threshold=self.contact_force_threshold
+        )
+        local_zmp_right, _ = evaluate_local_zmp(
+            wrench=right_wrench, contact_force_threshold=self.contact_force_threshold
+        )
+        com_from_desired = self.kindyn.getCenterOfMassPosition().toNumPy()
+        com_from_measured = self.kindyn_with_measured.getCenterOfMassPosition().toNumPy()
+
+        data = self.port.prepare()
+        data.vectors = {
+            "global_zmp": global_zmp,
+            "global_zmp_from_measured": global_zmp_from_measured,
+            "local_zmp_left": local_zmp_left,
+            "local_zmp_right": local_zmp_right,
+            "com_from_desired": com_from_desired,
+            "com_from_measured": com_from_measured,
+            "desired_torque": self.tsid.solver.get_output().joint_torques,
+        }
+        self.port.write()
+
+        if self.index * self.dt >= self.motion_duration + self.motion_timeout:
+            if self.knot_index + 1 >= len(self.com_knots_delta_x):
+                blf.log().info("Motion completed. Closing.")
+                return False
+
+            self.spline = self.create_new_spline(
+                [
+                    self.initial_com_position
+                    + np.array(
+                        [
+                            self.com_knots_delta_x[self.knot_index],
+                            self.com_knots_delta_y[self.knot_index],
+                            self.com_knots_delta_z[self.knot_index],
+                        ]
+                    ),
+                    self.initial_com_position
+                    + np.array(
+                        [
+                            self.com_knots_delta_x[self.knot_index + 1],
+                            self.com_knots_delta_y[self.knot_index + 1],
+                            self.com_knots_delta_z[self.knot_index + 1],
+                        ]
+                    ),
+                ],
+                self.motion_duration,
+                self.dt,
+            )
+
+            self.knot_index += 1
+            self.index = 0
+        else:
+            self.index += 1
+
+        return True
+
+
+    def __del__(self):
+        print("chiudo tutto e devo mori")
+
+        # switch the control mode to position control
+        if self.robot_control.is_valid():
+            self.robot_control.set_control_mode(
+                blf.robot_interface.YarpRobotControl.Position
+            )
+
+        # close the port
+        if not self.port.is_closed():
+            self.port.close()
+
+
+def main():
+    # Before everything let use the YarpSink for the logger and the YarpClock as clock. These are functionalities
+    # exposed by blf.
+    # if not blf.text_logging.LoggerBuilder.set_factory(
+    #     blf.text_logging.YarpLoggerFactory("balancing-torque-control")
+    # ):
+    #     blf.log().error("Impossible to set the logger factory.")
+    #     return False
```

Comment:
I do not know if you want to re-enable the logging. If not, you may cancel out these lines.

Reply:
I would say yes!